reviewing/dependencies-hierarchy/src/index.ts

```diff
@@ -302,16 +318,28 @@ function getTreeHelper (
       if (circular) {
         dependencies = []
       } else {
-        dependencies = dependenciesCache.get(packageAbsolutePath)
+        const requestedDepth = opts.maxDepth - opts.currentDepth
```

Comment:
The separate `maxDepth` and `currentDepth` options may make the recursion here harder to reason about. Instead of passing `currentDepth + 1` on each recursive call, how do we feel about passing `maxDepth - 1`? That removes an extra option here and simplifies the math a bit.

Reply:
Implementing at https://github.com/pnpm/pnpm/pull/5858.